File: backend/src/main/java/com/romeo/VetLink/user/UserDTO.java
package com.romeo.VetLink.user;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {
    private Integer id;

    private String firstName;

    private String lastName;

    private String email;

    private String password;

    @Enumerated(EnumType.STRING)
    private Role role;
}

File: frontend/src/app/auth/domain/user-dto.ts
import {UserRole} from "./user-role";

export class UserDTO {
   id : number | null = null;
   firstName: string | null = null;
   lastName: string | null = null;
   email: string | null = null;
   password: string | null = null;
   role: UserRole | null = null
}
